engine/engine.go

```diff
@@ -180,17 +183,15 @@ func (eng *Engine) Shutdown() {
 
 	// Call shutdown handlers, if any.
 	// Timeout after 10 seconds.
-	var wg sync.WaitGroup
 	for _, h := range eng.onShutdown {
-		wg.Add(1)
 		go func(h func()) {
-			defer wg.Done()
 			h()
+			eng.shutdownWait.Done()
 		}(h)
 	}
 	done := make(chan struct{})
 	go func() {
-		wg.Wait()
+		eng.shutdownWait.Wait()
 		close(done)
 	}()
 	select {
```

Comment:
now that we have the `shutdownWait` group, and we are waiting on it in the goroutine, do we still want this other channel with 10s timeout? I mean, behavior-wise I guess it is exactly the same for this sequence of actions, as the `eng.shutdownWait` replaces `wg`; but in the broader sense, the couple of PRs here have tried to make shutdown much more sane/controlled, and I wonder if that timeout is too short for the case of a significantly loaded system, maybe running devmapper, and doing a lot of cleanup?